src/integrations/GoogleCloudPubSubIntegration.ts

```diff
@@ -0,0 +1,336 @@
+
+        function pullWrapper(original: any, wrappedFunctionName: string) {
+            integration.wrappedFuncs[wrappedFunctionName] = original;
+            return function internalWrapper(request: any, options: any, callback: any) {
+
+                let span: ThundraSpan;
+                let reachedToCallOriginalFunc: boolean = false;
+                try {
+                    ThundraLogger.debug('<GoogleCloudPubSubIntegration> Tracing GoogleCloudPubSubIntegration pull');
+
+                    const { tracer } = ExecutionContextManager.get();
+
+                    if (!tracer) {
+                        ThundraLogger.debug('<GoogleCloudPubSubIntegration> Skipped tracing pull as no tracer is available');
+                        return original.apply(this, [request, options, callback]);
+                    }
+
+                    const originalOptions = typeof options !== 'function' ? options : undefined;
+                    const originalCallback = typeof options === 'function' ? options : callback;
+
+                    const originalFunction = integration.getOriginalFunction(wrappedFunctionName);
+                    const parentSpan = tracer.getActiveSpan();
+
+                    const subscription = request.subscription;
+                    const operationName = `PULL-${subscription}`;
+
+                    ThundraLogger.debug(`<GoogleCloudPubSubIntegration> Starting pull span with name ${operationName}`);
+
+                    const getMessageBody = (receivedMessages: any []) => {
+                        if (!receivedMessages) {
+                            return;
+                        }
+
+                        const messages = receivedMessages.map((receivedMessage) => receivedMessage.message);
+                        return GooglePubSubUtils.parseMessages(messages);
+                    };
+
+                    const getProjectId = () => {
+                        if (!subscription) {
+                            return;
+                        }
+
+                        const topicInfo = subscription.split('/');
+                        if (topicInfo.lenght < 1) {
+                            return;
+                        }
+
+                        return topicInfo[1];
+                    };
+
+                    span = tracer._startSpan(operationName, {
+                        childOf: parentSpan,
+                        domainName: DomainNames.API,
+                        className: ClassNames.GOOGLEPUBSUB,
+                        disableActiveStart: true,
+                    });
+
+                    const tags = {
+                        [GooglePubSubTags.PROJECT_ID]: getProjectId(),
+                        [GooglePubSubTags.SUBSCRIPTION]: subscription,
+                        [SpanTags.OPERATION_TYPE]: GooglePubSubOperationTypes.PULL,
+                        [SpanTags.SPAN_TYPE]: SpanTypes.GOOGLEPUBSUB,
+                        [SpanTags.TRACE_LINKS]: [span.spanContext.spanId],
```

Comment:
It might be better not to set `TRACE_LINKS` tag here as this is consumer side and no need to add an outgoing trace link (P.S. trace links set to span are added as outgoing trace link to invocation)